datafusion/core/src/datasource/physical_plan/file_scan_config.rs

```diff
@@ -101,6 +133,99 @@ pub struct FileScanConfig {
 }
 
 impl FileScanConfig {
+    /// Create a new `FileScanConfig` with default settings for scanning files.
+    ///
+    /// No file groups are added by default. See [`Self::add_file`] and
+    /// [`Self::add_file_group`]
+    ///
+    /// # Parameters:
+    /// * `object_store_url`: See [`Self::object_store_url`]
+    /// * `file_schema`: See [`Self::file_schema`]
+    pub fn new(object_store_url: ObjectStoreUrl, file_schema: SchemaRef) -> Self {
+        let statistics = Statistics::new_unknown(&file_schema);
+        Self {
+            object_store_url,
+            file_schema,
+            file_groups: vec![],
+            statistics,
+            projection: None,
+            limit: None,
+            table_partition_cols: vec![],
+            output_ordering: vec![],
+        }
+    }
+
+    /// Add a new file as a single file group
+    ///
+    /// See [Self::file_groups] for more information
+    pub fn add_file(&mut self, file: PartitionedFile) {
+        self.add_file_group(vec![file])
+    }
+
+    /// Add a new file group
+    ///
+    /// See [Self::file_groups] for more information
+    pub fn add_file_group(&mut self, file_group: Vec<PartitionedFile>) {
+        self.file_groups.push(file_group);
+    }
+
+    /// Set the statistics of the files
+    pub fn with_statistics(mut self, statistics: Statistics) -> Self {
+        self.statistics = statistics;
+        self
+    }
+
+    /// Set the projection of the files
+    pub fn with_projection(mut self, projection: Option<Vec<usize>>) -> Self {
+        self.projection = projection;
+        self
+    }
+
+    /// Set the limit of the files
+    pub fn with_limit(mut self, limit: Option<usize>) -> Self {
+        self.limit = limit;
+        self
+    }
+
+    /// Add a file as a single group
```

Comment:
Having to wrap a single file in a `vec![vec![..]]` makes sense from the implementation point of view, but most users shouldn't have to worry about this